### ISSUE TITLE
Text sometimes Draws at Wrong Location when doing Render to Texture

### ISSUE THREAD

**Issue by bradedelman** [pixijs/pixijs#8296]
### Expected Behavior
left and right output look the same

### Current Behavior
rendering on the right-side has text in the wrong place
![image](https://user-images.githubusercontent.com/124367/164291213-ed0badfa-5625-4423-8301-629fa210334d.png)

### Possible Solution
this happens with Text, but not Graphic or Sprite, so appears to be a bug with PIXI.Text
experimentally, if in Text.updateTexture, I remove the call to this._recursivePostUpdateTransform(); it works, but I don't think that is a correct fix.  somehow, the translation of a parent is applied when it shouldn't be.

### Steps to Reproduce
Basically, just text in a container with a parent that isn't at 0, 0 and then render that container to a Texture.
live playground with the code (<50 lines)
https://www.pixiplayground.com/#/edit/A9i0edGpz2lNDbARzQwe3

### Environment
- PIXI.JS 6.2.2
- Browser Chrome Version 100.0.4896.127 (Official Build) (arm64)
- OS MacOS 12.0
- Running Example https://www.pixiplayground.com/#/edit/A9i0edGpz2lNDbARzQwe3

**Comment by bigtimebuddy** [comment]
Thanks for reporting @bradedelman. This does seem like a bug, and seems to have been an issue for awhile.

**Comment by bradedelman** [comment]
fwiw, Text seems to be a Sprite subclass that manages its own texture.  Sprite never calls _recursivePostUpdateTransform itself, but Text does.  That said, DispalyObject calls it in a few places.  It seems to me that _recursivePostUpdateTransform fails to consider the render to texture case properly (where the full transform hierarchy may not be relevant).  That said, I'm unclear whether there are also cases where Sprite/DisplayObject will fail.  The thing that makes this hard to understand and fix is my lack of awareness of the overall architecture/strategy.  

That said, my naive question as a conversation starter to hope to lead to a fix: Why does Text need to call _recursivePostUpdateTransform, if Sprite does not?

**Comment by JetLua** [comment]
You may have to wait a step to do the calculation.

workaround: 
```js
app.renderer.once('postrender', () => app.renderer.render(container, { renderTexture: renderTexture }))
```

**Comment by dev7355608** [comment]
I'm not sure why `_recursivePostUpdateTransform` is even called here. `_onTextureUpdate` potentially updates the transform of the text: so the transforms of the children of the text are invalid after. I don't see the need to update the transforms of the parents. But I think `updateTexture` shouldn't do any transform updates at all, or `updateText` should not be called in `_render`: the transforms should not be changed while rendering. If the text is re-rendered and only the resolution is changed, it should be fine, because the size of the texture isn't affected by that, and therefore the transforms aren't changed. But if the more than the resolution was changed, the text should be updated in `updateTransform`, should it not?

To answer the question why the text is rendered at the wrong location, we have to look at this part in `Renderer.render`:
```ts
        if (!skipUpdateTransform)
        {
            // update the scene graph
            const cacheParent = displayObject.enableTempParent();

            displayObject.updateTransform();
            displayObject.disableTempParent(cacheParent);
            // displayObject.hitArea = //TODO add a temp hit area
        }
```
After the transforms are updated with a temporary parent, the original parent is restored. After that the object is rendered. But since the original parent is already restored, `_recursivePostUpdateTransform` goes beyond the root node of the subtree we want to render all the way to the root node of the entire tree. This breaks the transforms, and that's why the text in the example if offset by the same amount the outer container is offset. This problem does not only affect texts, but also `getBounds`/`toLocal`/`toGlobal` if called with `skipUpdate=false` after `updateTransform`. The original parent must be restored after rendering has completed.

**Comment by bradedelman** [comment]
@dev7355608 thanks for that information.  Using it I worked around the bug like this:

    // before the render, fix the text transforms
    const saveParent = container.enableTempParent();
    container.updateTransform();
    
    app.renderer.render(container, { renderTexture: renderTexture });

    // after the render, restore the parent/transforms
    container.disableTempParent(saveParent);
    container.updateTransform();
    
given your analysis, are you suggesting that removing the call to _recursivePostUpdateTransform is a correct fix?
